index.js

```diff
@@ -371,50 +370,75 @@ for (const el of helpers) {
 	got.stream[el] = (url, opts) => got.stream(url, Object.assign({}, opts, {method: el}));
 }
 
-function stdError(error, opts) {
-	if (error.code !== undefined) {
-		this.code = error.code;
-	}
+class StdError extends Error {
+	constructor(message, error, opts) {
+		super(message);
+		this.name = 'StdError';
 
-	Object.assign(this, {
-		message: error.message,
-		host: opts.host,
-		hostname: opts.hostname,
-		method: opts.method,
-		path: opts.path,
-		protocol: opts.protocol,
-		url: opts.href
-	});
-}
+		if (error.code !== undefined) {
+			this.code = error.code;
+		}
 
-got.RequestError = createErrorClass('RequestError', stdError);
-got.ReadError = createErrorClass('ReadError', stdError);
-got.ParseError = createErrorClass('ParseError', function (e, statusCode, opts, data) {
-	stdError.call(this, e, opts);
-	this.statusCode = statusCode;
-	this.statusMessage = http.STATUS_CODES[this.statusCode];
-	this.message = `${e.message} in "${urlLib.format(opts)}": \n${data.slice(0, 77)}...`;
-});
+		Object.assign(this, {
+			host: opts.host,
+			hostname: opts.hostname,
+			method: opts.method,
+			path: opts.path,
+			protocol: opts.protocol,
+			url: opts.href
+		});
+	}
+}
 
-got.HTTPError = createErrorClass('HTTPError', function (statusCode, headers, opts) {
-	stdError.call(this, {}, opts);
-	this.statusCode = statusCode;
-	this.statusMessage = http.STATUS_CODES[this.statusCode];
-	this.message = `Response code ${this.statusCode} (${this.statusMessage})`;
-	this.headers = headers;
-});
+got.RequestError = class RequestError extends StdError {
```

Comment:
You don't need to name the class. The JS engine is able to infer it.

```js
got.RequestError = class extends StdError {
```

Same with the others.